### PR TITLE
feat(react): added copy code functionality to `Code` component

### PR DESCRIPTION
**Pull request by denysfedorov**
Closes: https://github.com/dequelabs/cauldron/issues/1140

### CODE REVIEW

File: packages/react/src/components/Code/Code.test.tsx
Comment (scurker) [inline packages/react/src/components/Code/Code.test.tsx:20]: This globally overrides `navigator.clipBoard` so could affect other tests. You might not need this `Object.assign` once #1603 gets merged as this gets stubbed in `setupTests` and then you should be able to easily set a jest spy:

https://github.com/dequelabs/cauldron/blob/dc285a99a27a70b7b1bf62a8066d2acd6317f297/packages/react/src/setupTests.ts#L24-L32

https://github.com/dequelabs/cauldron/blob/dc285a99a27a70b7b1bf62a8066d2acd6317f297/packages/react/src/components/CopyButton/CopyButton.test.tsx#L38-L41

File: packages/react/src/components/Code/index.tsx
Comment (scurker) [inline packages/react/src/components/Code/index.tsx:85]: The id here should be generated, not static.

There's an example of how to implement this on Combobox with `useId`: https://github.com/dequelabs/cauldron/blob/f1f016388543360dc6a525037d583dc3b8f72835/packages/react/src/components/Combobox/Combobox.tsx#L420-L427

File: packages/react/src/components/Code/index.tsx
Comment (scurker) [inline packages/react/src/components/Code/index.tsx:84]: This would better follow our [naming conventions](https://cauldron.dequelabs.com/style-guidelines#styles):

```suggestion
        <div className="Code__Header">
```

File: packages/react/src/components/Code/Code.test.tsx
Comment (scurker) [inline packages/react/src/components/Code/Code.test.tsx:2]: [`user-event`](https://testing-library.com/docs/user-event/intro/) should be used over `fireEvent` when possible.

**Review by scurker — COMMENTED** [review]
I realize this is in draft, but did notice a few early things that need to be addressed.

File: packages/react/src/components/Code/index.tsx
Comment (denysfedorov) [reply packages/react/src/components/Code/index.tsx:85]: Fixed, create dynamic id for label

**Comment by scurker** [comment]
@denysfedorov #1603 was just merged, so this should be unblocked now

File: packages/react/src/components/Code/index.tsx
Comment (scurker) [inline packages/react/src/components/Code/index.tsx:29]: We generally don't rely on the export types:

```suggestion
  copyButtonProps?: React.ComponentProps<typeof CopyButtonProps>;
```

File: docs/pages/components/Code.mdx
Comment (scurker) [inline docs/pages/components/Code.mdx:112]:
```suggestion
      description: 'Prop to add CopyButton that copies code to the clipboard.'
```

File: packages/react/src/components/Code/index.tsx
Comment (scurker) [inline packages/react/src/components/Code/index.tsx:83]: The extra div is not necessary.

```suggestion
            <span id={`${id}-label`}>{label}</span>
```

File: packages/styles/code.css
Comment (scurker) [inline packages/styles/code.css:17]: The designs only show an `8px` spacing on the bottom between the header/code block.

File: packages/styles/code.css
Comment (scurker) [inline packages/styles/code.css:18]: There should not be a border on the header.

```suggestion
```

File: packages/react/src/components/Code/index.tsx
Comment (scurker) [inline packages/react/src/components/Code/index.tsx:82]: This should be `--font-weight-medium` and use `--text-size-body`.

File: packages/react/src/components/Code/Code.test.tsx
Comment (scurker) [inline packages/react/src/components/Code/Code.test.tsx:104]: We typically use `screen` because we want to ensure that components render to the document and not some unmounted container. This should be fixed for this test and the below tests.

**Review by scurker — CHANGES_REQUESTED** [review]
Just a few changes needed

File: packages/styles/code.css
Comment (denysfedorov) [reply packages/styles/code.css:17]: Added `margin-bottom: var(--space-smallest);` for spacing

File: docs/pages/components/Code.mdx
Comment (scurker) [inline docs/pages/components/Code.mdx:101]: This is actually the label for the heading and not the button. The button's name gets set via `CopyButtonProps`

```suggestion
      description: 'Label for the code heading'
```

File: docs/pages/components/Code.mdx
Comment (scurker) [inline docs/pages/components/Code.mdx:74]: Just give this something a little more descriptive:

```suggestion
<Code allowCopy label="Example Code Heading">
```

**Review by scurker — CHANGES_REQUESTED** [review]
Code wise, this looks good aside from the 2 remaining changes that need to be made.

Can someone from @dequelabs/cauldron-design also give this a design review?

**Comment by shawnsharpDQ** [comment]
@denysfedorov could you look at the example button in the dark theme? When I hover over it I notice a movement, I don't see it on any other button. 
Example copy button on dark

**Comment by denysfedorov** [comment]
> @denysfedorov could you look at the example button in the dark theme? When I hover over it I notice a movement, I don't see it on any other button.
@shawnsharpDQ @scurker 

I believe this is because of border that is attached to the `.cauldron--theme-dark .Button--tertiary:is(:hover, :active, :focus)`. But I can't explain why we have such behaviour only for dark theme. @scurker  may I ask you about this

https://github.com/user-attachments/assets/2fade0ea-743b-43ba-860b-98eccfc5da41

**Comment by scurker** [comment]
It looks like it's an issue with the base copy button in dark mode, I see it on dev: https://cauldron.dequelabs.com/components/CopyButton

@shawnsharpDQ @denysfedorov Since this isn't an individual problem with @denysfedorov's PR - can we raise an issue to be fixed separately and unblock this PR?

**Comment by scurker** [comment]
@denysfedorov The only thing blocking this PR is having an issue created for the above comment, as soon as the issue is there we can unblock/merge.

**Comment by denysfedorov** [comment]
> @denysfedorov The only thing blocking this PR is having an issue created for the above comment, as soon as the issue is there we can unblock/merge.

Created a new issue for base copy button
https://github.com/dequelabs/cauldron/issues/1659

**Review by scurker — APPROVED** [review]

**Pull request merged**
